sgx-tools/dcap_linux.go

```diff
@@ -0,0 +1,514 @@
+	}
+
+	if resp.GetQuoteExRes.GetQeReportInfo() != nil {
+		return nil, fmt.Errorf("invalid Qe Rereport Info:%v\n", resp.GetQuoteExRes.GetQeReportInfo())
+	}
+
+	return resp.GetQuoteExRes.GetQuote(), nil
+}
+
+func selectAttKeyIDRequest() ([]byte, error) {
+	conn, err := dialAesmd()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	req := pb.AesmServiceRequest{}
+	req.SelectAttKeyIDReq = &pb.AesmServiceRequest_SelectAttKeyIDRequest{
+		Timeout: timeOut,
+	}
+
+	rdata, err := transmitAesmd(conn, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.SelectAttKeyIDRes = &pb.AesmServiceResponse_SelectAttKeyIDResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return nil, err
+	}
+
+	if resp.SelectAttKeyIDRes.GetErrorCode() != 0 {
+		return nil, fmt.Errorf("failed to select Att Key(error code = %d)",
+			resp.SelectAttKeyIDRes.GetErrorCode())
+	}
+
+	if resp.SelectAttKeyIDRes.GetSelectedAttKeyId() == nil {
+		return nil, fmt.Errorf("failed to select Att key Id")
+	}
+
+	return resp.SelectAttKeyIDRes.GetSelectedAttKeyId(), nil
+}
+
+func getSupportedAttKeyIDNumRequest() (uint32, error) {
+	conn, err := dialAesmd()
+	if err != nil {
+		return 0, err
+	}
+	defer conn.Close()
+
+	req := pb.AesmServiceRequest{}
+	req.GetSupportedAttKeyIDNumReq = &pb.AesmServiceRequest_GetSupportedAttKeyIDNumRequest{
+		Timeout: timeOut,
+	}
+
+	rdata, err := transmitAesmd(conn, &req)
+	if err != nil {
+		return 0, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.GetSupportedAttKeyIDNumRes = &pb.AesmServiceResponse_GetSupportedAttKeyIDNumResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return 0, err
+	}
+
+	if resp.GetSupportedAttKeyIDNumRes.GetErrorCode() != 0 {
+		return 0, fmt.Errorf("failed to get Supported Att Key ID Num (error code = %d)",
+			resp.GetSupportedAttKeyIDNumRes.GetErrorCode())
+	}
+
+	if resp.GetSupportedAttKeyIDNumRes.GetAttKeyIdNum() == 0 {
+		return 0, fmt.Errorf("invalid Att key Id Num = %d",
+			resp.GetSupportedAttKeyIDNumRes.GetAttKeyIdNum())
+	}
+
+	return resp.GetSupportedAttKeyIDNumRes.GetAttKeyIdNum(), nil
+}
+
+func getSupportedAttKeyIDsRequest(num uint32) ([]byte, error) {
+	conn, err := dialAesmd()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	req := pb.AesmServiceRequest{}
+	req.GetSupportedAttKeyIDsReq = &pb.AesmServiceRequest_GetSupportedAttKeyIDsRequest{
+		BufSize: num * attestationKeyIdentityLength,
+		Timeout: timeOut,
+	}
+
+	rdata, err := transmitAesmd(conn, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.GetSupportedAttKeyIDsRes = &pb.AesmServiceResponse_GetSupportedAttKeyIDsResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return nil, err
+	}
+
+	if resp.GetSupportedAttKeyIDsRes.GetErrorCode() != 0 {
+		return nil, fmt.Errorf("failed to get Supported Att Key IDs List (error code = %d)",
```

Comment:
Change to
```
failed to call GetSupportedAttKeyIDsRequest (error code = %d)
```

All error messages in pb wrapper functions follow this style.

The error messages occurring in wrapper functions should use